builtin/common/tests/vector_spec.lua

```diff
@@ -38,9 +63,45 @@ describe("vector", function()
 			local a = { x = 2, y = 4, z = -10 }
 			assertE(a, a)
 			assertNE({x = -1, y = 0, z = 1}, a)
+
+			assert.equal(vector.new(1, 2, 3), vector.new(1, 2, 3))
 	end)
 
 	it("add()", function()
-		assert.same({ x = 2, y = 4, z = 6 }, vector.add(vector.new(1, 2, 3), { x = 1, y = 2, z = 3 }))
+		assert.equal(vector.new(2, 4, 6), vector.add(vector.new(1, 2, 3), { x = 1, y = 2, z = 3 }))
```

Comment:
why does `same` not work here?

Reply:
Same seems to compare the raw tables.
(I don't think any mod does this / needs to do this.)

Reply:
Now that next and pairs is overridden, same would work here.

Reply:
But `equal` is fine. (`same` was *needed* in the past.)